src/test/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportControlCheckTest.java

```diff
@@ -182,4 +182,45 @@ public void testWrongFormatURI() throws Exception {
                 + "InputImportControl.java"), expected);
     }
 
+    @Test
+    public void testUrl() throws Exception {
+        final DefaultConfiguration checkConfig = createCheckConfig(ImportControlCheck.class);
+        checkConfig.addAttribute("url", getURI("imports" + File.separator
+                + "import-control_one.xml"));
+        final String[] expected = {"5:1: " + getCheckMessage(MSG_DISALLOWED, "java.io.File")};
+
+        verify(checkConfig, getPath("imports" + File.separator
+                + "InputImportControl.java"), expected);
+    }
+
+    @Test
+    public void testUrlBlank() throws Exception {
+        final DefaultConfiguration checkConfig = createCheckConfig(ImportControlCheck.class);
+        checkConfig.addAttribute("url", "");
+        final String[] expected = {"1:40: " + getCheckMessage(MSG_MISSING_FILE)};
+
+        verify(checkConfig, getPath("imports" + File.separator
+                + "InputImportControl.java"), expected);
+    }
+
+    @Test(expected = CheckstyleException.class)
```

Comment:
Yes, this code is a bit better, but the main problem still the same: analysis of exception message doesn't makes sense, it has no context of original problem. I receive two identical messages for two different problems.